### PR TITLE
CLOUDP-153344: Fix network peering

### PR DESCRIPTION
**Pull request by Sugar-pack**
## Proposed changes

_Jira ticket:_ CLOUDP-153344

## Checklist

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added any necessary documentation in document requirements section listed in [CONTRIBUTING.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/CONTRIBUTING.md) (if appropriate)
- [ ] I have addressed the @mongodb/docs-cloud-team comments (if appropriate)
- [x] I have updated [test/README.md](https://github.com/mongodb/mongodb-atlas-cli/blob/master/test/README.md) (if an e2e test has been added)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: internal/kubernetes/operator/project/project.go [hunk 1]
```diff
@@ -394,41 +394,76 @@ func buildPrivateEndpoints(peProvider store.PrivateEndpointLister, projectID str
...
-	npList, err := npProvider.PeeringConnections(projectID, &atlas.ContainersListOptions{
+	npListAWS, err := npProvider.PeeringConnections(projectID, &atlas.ContainersListOptions{
 		ListOptions: atlas.ListOptions{
 			ItemsPerPage: MaxItems,
 		},
+		ProviderName: string(provider.ProviderAWS),
 	})
 	if err != nil {
-		return nil, err
+		return nil, fmt.Errorf("error getting network peering connections for AWS: %s", err)
```
Comment (gssbzn) [inline internal/kubernetes/operator/project/project.go:405]: let's use improved error handling for go 
```suggestion
		return nil, fmt.Errorf("error getting network peering connections for AWS: %w", err)
```

File: internal/kubernetes/operator/project/project.go [hunk 2]
```diff
@@ -394,41 +394,76 @@ func buildPrivateEndpoints(peProvider store.PrivateEndpointLister, projectID str
...
-			NetworkName:         np.NetworkName,
-		})
+	npListGCP, err := npProvider.PeeringConnections(projectID, &atlas.ContainersListOptions{
+		ListOptions: atlas.ListOptions{
+			ItemsPerPage: MaxItems,
+		},
+		ProviderName: string(provider.ProviderGCP),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting network peering connections for GCP: %s", err)
```
Comment (gssbzn) [inline internal/kubernetes/operator/project/project.go:415]:
```suggestion
		return nil, fmt.Errorf("error getting network peering connections for GCP: %w", err)
```

File: internal/kubernetes/operator/project/project.go [hunk 3]
```diff
@@ -394,41 +394,76 @@ func buildPrivateEndpoints(peProvider store.PrivateEndpointLister, projectID str
...
+	}
+
+	npListAzure, err := npProvider.PeeringConnections(projectID, &atlas.ContainersListOptions{
+		ListOptions: atlas.ListOptions{
+			ItemsPerPage: MaxItems,
+		},
+		ProviderName: string(provider.ProviderAzure),
+	})
+	if err != nil {
+		return nil, fmt.Errorf("error getting network peering connections for Azure: %s", err)
```
Comment (gssbzn) [inline internal/kubernetes/operator/project/project.go:425]:
```suggestion
		return nil, fmt.Errorf("error getting network peering connections for Azure: %w", err)
```

File: internal/kubernetes/operator/project/project_test.go [hunk 4]
```diff
@@ -12,7 +12,7 @@
 // See the License for the specific language governing permissions and
 // limitations under the License.
 
-//go:build unit
+///go:build unit
```
Comment (gssbzn) [inline internal/kubernetes/operator/project/project_test.go:15]:
```suggestion
//go:build unit
```

File: internal/kubernetes/operator/project/project.go (hunk 1 above)
Comment (Sugar-pack) [reply internal/kubernetes/operator/project/project.go]: Maybe it makes sense to add [errorlint](https://github.com/polyfloyd/go-errorlint) to golangci-lint.

Comment (gssbzn) [reply internal/kubernetes/operator/project/project.go]: go for it, for conext we started with go1.12 when I tried to enable some of those error linters there was a lot of technical debt being flagged, not sure what the state is right now as I've been trying to fix them over time

**Review by gssbzn — APPROVED** [review]
LGTM

**Pull request merged**